sklearn/utils/_indexing.py

```diff
@@ -417,7 +417,14 @@ def _get_column_indices_interchange(X_interchange, key, key_dtype):
     },
```

Comment:
This comment has still not been addressed.

Reply:
The new parameter should also be documented in the docstring since this is a public function.

Also, we could expand the documentation for the `replace` parameter to state that it is recommended to use `replace=True` whenever sampling with non-uniform weights: a few data points with very large weights are expected to be sampled several times with probability to preserve the distribution induced by the weights.